bench/service_bench.exs

```diff
@@ -0,0 +1,56 @@
+defmodule EchoService do
```

Comment:
I added a benchmark to try to measure the impact of these changes, but it turns out this branch only make a ~5% improvement in throughput.

We can get around 13.7k service messages per second on both the main branch and about 14.5k service messages per second on this branch. 🤷